corehq/apps/app_manager/tests/test_suite_registry_search.py

```diff
@@ -216,6 +221,61 @@ def test_form_linking_to_registry_module_from_registration_form(self):
             "./entry[2]/stack/create"
         )
 
+    def test_workflow_registry_module_previous_screen_after_case_list_form(self):
+        factory = AppFactory(DOMAIN, "App with DR", build_version='2.53.0')
+        m0, f0 = factory.new_basic_module("new case", "case")
+        factory.form_opens_case(f0, "case")
+
+        m1, f1 = factory.new_basic_module("update case", "case")
+        factory.form_requires_case(f1, "case")
+
+        m1.case_details.long.columns.append(
+            DetailColumn.wrap(dict(
+                header={"en": "name"},
+                model="case",
+                format="plain",
+                field="whatever",
+            ))
+        )
+
+        m1.search_config = CaseSearch(
+            properties=[
+                CaseSearchProperty(name='name', label={'en': 'Name'}),
+            ],
+            data_registry="myregistry",
+            data_registry_workflow=REGISTRY_WORKFLOW_LOAD_CASE,
+        )
+
+        m1.case_list_form.form_id = f0.get_unique_id()
+        m1.case_list_form.label = {'en': 'New Case'}
+
+        f1.post_form_workflow = WORKFLOW_PREVIOUS
+        app = Application.wrap(factory.app.to_json())
+        app._id = "123"
+        suite = app.create_suite()
+        self.assertXmlPartialEqual(
+            """
+            <partial>
+              <create>
+                <command value="'m1'"/>
+                <query id="results" value="http://localhost:8000/a/test_domain/phone/registry_case/123/">
+                  <data key="case_type" ref="'case'"/>
+                  <data key="x_commcare_data_registry" ref="'myregistry'"/>
+                  <data key="case_id" ref="instance('commcaresession')/session/data/case_id"/>
```

Comment:
prior to this PR the ref here would have been `instance('commcaresession')/session/data/case_id_new_case_0`